wasmplugin/body.go

```diff
@@ -0,0 +1,50 @@
+// Copyright The OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package wasmplugin
+
+import (
+	"io"
+
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+)
+
+type ByteLenger interface {
+	Len() int
+}
+
+// bodyWrapper is a wrapper around the GetHttpRequestBody which exposes
+// the length of the body on beforehand so the tx.ReadRequestBodyFrom can
+// look at the body length first before attempting to copy the bytes.
+type bodyWrapper struct{ totalSize int }
+
+var (
+	_ io.Reader  = bodyWrapper{}
+	_ ByteLenger = bodyWrapper{}
+)
+
+func (w bodyWrapper) Read(p []byte) (int, error) {
+	var (
+		readingBytes int
+		lenErr       error
+	)
+
+	if len(p) > readingBytes {
```

Comment:
Think this is `len(p) > w.totalSize`? `readingBytes` is always `0`

Reply:
I think the simpler approach is probably still fine, the body method is only called with each HTTP frame / chunk, not entire payload, but if going with this one confirm there is test coverage for the cases

Reply:
Yeah we need coverage on this, just trying the approach.